lib/touch-enroll.js

```diff
@@ -0,0 +1,95 @@
+import { exec as nodeExec } from 'child_process';
+import B from 'bluebird';
+
+// Had problems with teen_process, resorted to this
+const exec = B.promisify(nodeExec);
+
+let touchEnrollMenuKeys = ['Toggle Enrolled State', 'Touch ID Enrolled'];
+let touchEnrollBackups;
+const NS_USER_KEY_EQUIVALENTS = 'NSUserKeyEquivalents';
+const TOUCH_ENROLL_KEY_CODE = '@~$^t';
+
+async function setTouchEnrollKey () {
+  await backupTouchEnrollShortcuts();
+  for (let key of touchEnrollMenuKeys) {
+    await setUserDefault(NS_USER_KEY_EQUIVALENTS, key, TOUCH_ENROLL_KEY_CODE);
+  }
+}
+
+async function setTouchEnrollKeys (pairs) {
+  for (let [key, value] of pairs) {
+    await setUserDefault(NS_USER_KEY_EQUIVALENTS, key, value);
+  }
+}
+
+async function getTouchEnrollKeys () {
+  let backups = [];
+  for (let key of touchEnrollMenuKeys) {
+    backups.push([key, await getUserDefault(NS_USER_KEY_EQUIVALENTS, key)]);
+  }
+  return backups;
+}
+
+/**
+ * Get MacOS User Defaults by domain and key (for reference: `man defaults`)
+ * @param {*} domain {string}
+ * @param {*} key {string|number|boolean}
+ */
+async function getUserDefault (domain, key) {
+  let res;
+  try {
+    // If it doesn't find anything for this domain, it will throw an error so return undefined
+    res = await exec(`"defaults" read "Apple Global Domain" ${domain}`);
```

Comment:
`exec` is the promisified form of `child_process.exec`. This will be changed back to teen_process though.